vulcanai2/models/metrics.py

```diff
@@ -1,66 +1,371 @@
 import numpy as np
+
 import torch
 import torch.nn.functional as F
 from torch.autograd import Variable
-from sklearn import metrics as scipy_metrics
+
+import math
+import numpy as np
+from sklearn import metrics as skl_metrics
+from sklearn.metrics import confusion_matrix
+from sklearn.preprocessing import LabelBinarizer
+from sklearn.manifold import TSNE
+from sklearn.decomposition import PCA
+
+from copy import deepcopy
+
+from collections import Counter
+
 import logging
 logger = logging.getLogger(__name__)
 
 class Metrics(object):
 
-    def __init__(self, num_class):
+    def __init__(self, num_class, use_unlabeled=False):
+        self.mat = np.zeros((num_class, num_class), dtype=np.float)
+        self.valids = np.zeros((num_class), dtype=np.float)
+        self.IoU = np.zeros((num_class), dtype=np.float)
+        self.mIoU = 0
+
         self.num_class = num_class
-        self.confusion_matrix = np.zeros((num_class, num_class))
-
-    def _fast_hist(self, prediction, target, num_class):
-        mask = (target >= 0) & (target < num_class)
-        hist = np.bincount(
-                            num_class * target[mask].astype(int) + prediction[mask],
-                            minlength=num_class ** 2,
-                            ).reshape(num_class, num_class)
-        return hist
+        self.list_classes = list(range(num_class))
+        self.use_unlabeled = use_unlabeled
+        self.mat_start_idx = 1 if not self.use_unlabeled else 0
+
+    def reset(self):
```

Comment:
Why do we reset the metrics matrix's? why not just overwrite them?

Reply:
thought it would be modular having reset as an option and helpful while logging. Not a good idea?

Reply:
I mean... if you can think of a key case that requires resetting the metric matrices but I can't think of one. it may be redundant when the logger could overwrite with the most recent test. idk what did you plan on using it for?